packages/amplify-cli/src/extensions/amplify-helpers/resource-status-diff.ts

```diff
@@ -0,0 +1,295 @@
+    private getCfnResourceFilePaths = async() => {
+      return {
+            localTemplatePath : (checkExist(this.resourceFiles.localBuildCfnFile))?this.resourceFiles.localBuildCfnFile: this.resourceFiles.localPreBuildCfnFile ,
+            cloudTemplatePath : (checkExist(this.resourceFiles.cloudBuildCfnFile))?this.resourceFiles.cloudBuildCfnFile: this.resourceFiles.cloudPreBuildCfnFile
+      }
+    }
+
+    //helper: Get category provider from filename
+    private normalizeProviderForFileNames(provider:string):string{
+      //file-names are currently standardized around "cloudformation" not awscloudformation.
+      if ( provider === "awscloudformation"){
+        return CategoryProviders.CLOUDFORMATION
+      } else {
+        return provider;
+      }
+    }
+
+    //helper: Convert cloudformation template diff data using CDK api
+    private printStackDiff = ( templateDiff, stream?: cfnDiff.FormatStream ) =>{
+        // filter out 'AWS::CDK::Metadata' resources from the template
+        if (templateDiff.resources ) {
+          templateDiff.resources = templateDiff.resources.filter(change => {
+            if (!change) { return true; }
+            if (change.newResourceType === 'AWS::CDK::Metadata') { return false; }
+            if (change.oldResourceType === 'AWS::CDK::Metadata') { return false; }
+            return true;
+          });
+        }
+        if (!templateDiff.isEmpty) {
+          cfnDiff.formatDifferences(stream || process.stderr, templateDiff);
+        }
+        return templateDiff.differenceCount;
+    }
+
+    //helper: reads file and deserializes
+    private loadStructuredFile = async(fileName: string, deserializeFunction) => {
+        const contents = await fs.readFile(fileName, { encoding: 'utf-8' });
+        return deserializeFunction(contents);
+    }
+
+    //helper: returns file-extention of the filePath
+    private getFilePathExtension( filePath :string ){
+       const ext = path.extname( filePath );
+       return (ext)? (ext.substring(1)).toLowerCase() : "" ;
+    }
+
+    //Search and return the path to cloudformation template in the given folder
+    private globCfnFilePath( fileFolder : string ){
+      if (fs.existsSync( fileFolder )) {
+        const globOptions: glob.IOptions = {
+          absolute: false,
+          cwd: fileFolder,
+          follow: false,
+          nodir: true,
+        };
+        const templateFileNames = glob.sync('**/*template.{yaml,yml,json}', globOptions);
+        for (const templateFileName of templateFileNames) {
+          const absolutePath = path.join(fileFolder, templateFileName);
+          return absolutePath; //We support only one cloudformation template ( nested templates are picked )
+        }
+      }
+      return ""
+    }
+
+    //Load Cloudformation template from file.
+    //The filePath should end in json/yaml or yml
+    private loadCfnTemplate = async(filePath) => {
```

Comment:
Should this be a common helper code and I think we have a common helper function for loading cfn file? cc @attilah @yuth

Reply:
Looks like there is a function readCFNTemplate in amplify-core . I will refactor to use that and remove this extraneous code.